### PR TITLE
Checking for local external storage flag iff the method is local.

### LINKED ISSUE #27619: Cannot create any external storage without local flag
### Steps

1. Setup OC from scratch
1. Setup a SFTP storage as admin

### Expected
Storage can be setup.

### Actual
403 forbidden exception.

### Version
ownCloud 10.0 beta2

It only works if `files_external_allow_create_new_local` is set to `true` in the config.

This flag is only supposed to prevent creation of **Local** storages.

@noveens do you have time to fix this critical regression ?

### PR DESCRIPTION
**Pull request by noveens**
Fixes #27619 

## Description

Regression from #27590 

## Related Issue

https://github.com/owncloud/core/issues/27619

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by noveens** [comment]
sorry for the silly mistake @PVince81 :stuck_out_tongue:

File: apps/files_external/lib/Controller/UserStoragesController.php [hunk 1]
```diff
@@ -128,38 +128,36 @@ public function create(
...
-			$this->updateStorageStatus($newStorage);
-
-			return new DataResponse(
-				$newStorage,
-				Http::STATUS_CREATED
-			);
-		}
-
-		else {
+		if ($backend === 'local' && $canCreateNewLocalStorage === false) {
```
Comment (PVince81) [inline apps/files_external/lib/Controller/UserStoragesController.php]: you could remove this completely, users are never allowed to setup local storages for security reasons

**Comment by noveens** [comment]
@PVince81 , done.

**Comment by PVince81** [comment]
Tested, works.

Please fix the unit tests:
```
19:35:53 1) OCA\Files_External\Tests\Controller\GlobalStoragesControllerTest::testAddStorageWithoutConfig
19:35:53 OCP\Files\External\Service\IStoragesService::createStorage('mount', '\OCA\Files_External\Lib\Storage\SMB', '\OCA\Files_External\Lib\Auth\...hanism', Array (), Array (), Array (), Array (), null) was not expected to be called.
19:35:53 
19:35:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-27621-QCHHOII2IVFMCUHAYJPIT6GBRSN6JJRAA5ZQSU6L5LKW7XNJQPCA@2/apps/files_external/lib/Controller/StoragesController.php:115
19:35:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-27621-QCHHOII2IVFMCUHAYJPIT6GBRSN6JJRAA5ZQSU6L5LKW7XNJQPCA@2/apps/files_external/lib/Controller/GlobalStoragesController.php:99
19:35:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-27621-QCHHOII2IVFMCUHAYJPIT6GBRSN6JJRAA5ZQSU6L5LKW7XNJQPCA@2/apps/files_external/tests/Controller/StoragesControllerTest.php:160
19:35:53 
19:35:53 2) OCA\Files_External\Tests\Controller\UserStoragesControllerTest::testAddStorageWithoutConfig
19:35:53 OCP\Files\External\Service\IStoragesService::createStorage('mount', '\OCA\Files_External\Lib\Storage\SMB', '\OCA\Files_External\Lib\Auth\...hanism', Array (), Array (), null, null, null) was not expected to be called.
19:35:53 
19:35:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-27621-QCHHOII2IVFMCUHAYJPIT6GBRSN6JJRAA5ZQSU6L5LKW7XNJQPCA@2/apps/files_external/lib/Controller/StoragesController.php:115
19:35:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-27621-QCHHOII2IVFMCUHAYJPIT6GBRSN6JJRAA5ZQSU6L5LKW7XNJQPCA@2/apps/files_external/lib/Controller/UserStoragesController.php:129
19:35:53 /var/lib/jenkins/workspace/owncloud-core_core_PR-27621-QCHHOII2IVFMCUHAYJPIT6GBRSN6JJRAA5ZQSU6L5LKW7XNJQPCA@2/apps/files_external/tests/Controller/StoragesControllerTest.php:160
```

**Comment by noveens** [comment]
@PVince81 ,
I have updated the test, and the test requires the check condition here as well : 
https://github.com/owncloud/core/pull/27621/files#diff-432f5e183c311067085bd0348a8191fcR131

**Comment by PVince81** [comment]
Tested, works :+1:

Let's wait for CI now

**Comment by noveens** [comment]
@PVince81 ,
Jenkins passed :smiley: 
Merge?

**Pull request merged**